### ISSUE TITLE
Incorrect scmsync tag set on branch_package step for SCM/CI integration

### ISSUE THREAD

**Issue by krauselukas** [openSUSE/open-build-service#13290]
# Issue Description

A [PR](https://github.com/SUSE/BCI-dockerfile-generator/pull/86) against the 'tumbleweed' branch with the following `workflow.yml` resulted in setting an incorrect `<scmsync>` tag in the meta of the branched package.

*workflow.yml:*
```
---
workflow:
  steps:
    - branch_package:
        source_project: home:dancermak:BCI:Tumbleweed
        source_package: 389-ds-container
        target_project: home:dancermak:BCI:Tumbleweed:Staging
```

Resulting package meta on [branched package](https://build.opensuse.org/package/meta/home:dancermak:BCI:Tumbleweed:Staging:SUSE:BCI-dockerfile-generator:PR-86/389-ds-container):
```
<package name="389-ds-container" project="home:dancermak:BCI:Tumbleweed:Staging:SUSE:BCI-dockerfile-generator:PR-86">
  <title/>
  <description/>
  <scmsync>https://github.com/SUSE/BCI-dockerfile-generator.git?subdir=389-ds-container#tumbleweed?subdir=389-ds-container#cc3a337b4682ae040016a4a11e59cc30d581a475</scmsync>
</package>
```

This is the incorrect syntax to point to the new commit using the scm bridge.
See https://openbuildservice.org/help/manuals/obs-user-guide/cha.obs.scm_bridge.html

Expected Result
============

The package meta of of the  [branched package](https://build.opensuse.org/package/meta/home:dancermak:BCI:Tumbleweed:Staging:SUSE:BCI-dockerfile-generator:PR-86/389-ds-container) should look like this:

```
<package name="389-ds-container" project="home:dancermak:BCI:Tumbleweed:Staging:SUSE:BCI-dockerfile-generator:PR-86">
  <title/>
  <description/>
  <scmsync>https://github.com/SUSE/BCI-dockerfile-generator.git?subdir=389-ds-container#cc3a337b4682ae040016a4a11e59cc30d581a475</scmsync>
</package>
```

How to Reproduce
===============
1. Create a package with the name `389-ds-container`
2. Add the `<scmsync>` of the failing example to the package meta and see it failing by not fetching the sources...
3. Modify the `<scmsync>` tag again, now exchanging it with the correct example. See it pulling the correct sources... (please be aware that the commit sha can change whenever someone pushed to the PR, so make sure you use the latest commit sha)

**Comment by dcermak** [comment]
tl;dr; the problem is that the obs CI does not properly recognize either the `subdir` or the fragment and just appends the new one resulting in the following bogus URL:
`https://github.com/SUSE/BCI-dockerfile-generator.git?subdir=389-ds-container#tumbleweed?subdir=389-ds-container#cc3a337b4682ae040016a4a11e59cc30d581a475`
(`subdir` and the fragment appear twice)
